### PR TITLE
Feature/global state

### PR DESCRIPTION
**Pull request by Cyanne-Jones**
What does this PR do?
- This branch added global state management, delete and favorite functionality
Where should your reviewer start?
- Open the project in the browser and try saving facts to the favorited array 
What (if any) features are you implementing?
- Pinia state management
What (if anything) did you refactor?
- Address merge conflicts
Were there any issues that arose?
- Nothing too bad!
Is there anything that you need from your teammate?
- 
Any other comments, questions, or concerns?
-

### CODE REVIEW

File: src/stores/boosterStore.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  }, 
+  actions: {
+    favoriteBooster(booster){
+      booster.isFavorited = true;
+      this.boosters.push(booster)
+    },
+    deleteBooster(deletedBooster){
+      deletedBooster.isFavorited = false;
+      this.boosters = this.boosters.filter(booster => booster.id !== deletedBooster.id)
+    }
```
Comment (masaki-kleinkopf) [inline src/stores/boosterStore.js:27]: This was great learning for us to figure out how to use pinia.

File: src/views/RandomView.vue [hunk 2]
```diff
@@ -49,20 +49,20 @@ import { ref } from "vue"
...
-        :title="booster.title"
-        :type="booster.type"
-        :preparation="booster.preparation"
-        :ingredients="booster.ingredients"
-        :isFavorited="booster.isFavorited"
-      />
-    </div>
+    
+    <BoosterCard v-if= "booster && booster.type ==='fact'" 
+      :isFavorited="booster.isFavorited"
```
Comment (davidhdaw) [inline src/views/RandomView.vue:54]: Do we actually need anything but booster as a prop?

File: src/views/SavedView.vue [hunk 3]
```diff
@@ -1,12 +1,42 @@
...
+    </div>
+    <div class="drink-container booster-container"
+    v-if="savedMode === 'drinks'"
+      v-for="drink in store.favoritedDrinks">
+      <DrinkBoosterCard 
+      :title="drink.title"
+      :type="drink.type"
+      :preparation="drink.preparation"
+      :ingredients="drink.ingredients"
+      :isFavorited="drink.isFavorited"
```
Comment (masaki-kleinkopf) [inline src/views/SavedView.vue:37]: Nice work displaying our saved cards.

File: src/stores/boosterStore.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { defineStore } from 'pinia'
+
+export const useBoosterStore = defineStore({
+  id: 'boosters',
+  state: () => ({
+    boosters: [],
+  }),
+  getters: {
+    favoritedFacts: (state) => {
```
Comment (davidhdaw) [inline src/stores/boosterStore.js:9]: I kinda like this better than computed values within a component.

**Review by davidhdaw — APPROVED** [review]

**Review by masaki-kleinkopf — APPROVED** [review]

**Pull request merged**